Gems/RecastNavigation/Code/CMakeLists.txt

```diff
@@ -49,12 +49,35 @@ set_property(
     PROPERTY COMPILE_OPTIONS ${compile_options}
 )
 unset(compile_options)
+
+# Place external recast libraries under RecastNavigation gem.
 set(RECAST_FETCHCONTENT_FOLDER "Gems/RecastNavigation/External")
```

Comment:
There shouldn't be an assumption that the gem resides underneath the O3DE repo root.
For each gem that is registered we provide a CMake GLOBAL property alias of "@gemroot:<gem-name>@" that can be used to look up this folder
`get_property(${NameLower}_gem_root GLOBAL PROPERTY "@GEMROOT:RecastNavigation@")`

Reply:
You can then use [cmake_path(RELATIVE_PATH](https://cmake.org/cmake/help/latest/command/cmake_path.html?highlight=cmake_path#generation) to create a relative path from the `${CMAKE_SOURCE_DIR}` to the this gem path in order to populate the Visual Studio folder entry

Reply:
Done.